numba/core/interpreter.py

```diff
@@ -1283,6 +1554,28 @@ def op_JUMP_IF_FALSE_OR_POP(self, inst, pred):
     def op_JUMP_IF_TRUE_OR_POP(self, inst, pred):
         self._op_JUMP_IF(inst, pred=pred, iftrue=True)
 
+    def op_JUMP_IF_NOT_EXC_MATCH(self, inst, pred, tos, tos1):
+        truebr = inst.next
+        falsebr = inst.get_jump_target()
+        gv_fn = ir.Global(
+            "exception_match", eh.exception_match, loc=self.loc,
+        )
+        exc_match_name = '$exc_match'
+        self.store(value=gv_fn, name=exc_match_name, redefine=True)
+        lhs = self.get(tos1)
+        rhs = self.get(tos)
+        exc = ir.Expr.call(
+            self.get(exc_match_name), args=(lhs, rhs), kws=(), loc=self.loc,
+        )
+        predicate = self.store(exc, pred)
+        bra = ir.Branch(cond=predicate, truebr=truebr, falsebr=falsebr,
+                        loc=self.loc)
+        self.current_block.append(bra)
+
+    def op_RERAISE(self, inst, exc):
+        # Do nothing, Numba can't handle this case and it's caught else where
+        pass
```

Comment:
Better to put in a `StaticRaise` here.
```suggestion
        self.current_block.append(ir.StaticRaise(None, None, loc=self.loc))
```

Reply:
It's really to guard against getting a RERAISE unexpectedly. The `StaticRaise` will terminate the execution, even thought it is not raising the correct exception.

Reply:
Should we perhaps also note that it's not doing the right thing as part of the error message in the `StaticRaise` ?

Reply:
Any idea if this is actually reachable during execution? This sort of thing:
```
@njit
def foo():
    try:
        raise Exception
    except Exception:
        raise ValueError("ERROR")
```
will create a `RERAISE` op code that actually gets interpreted, but it's not reachable at runtime. Most things I've tried follow the pattern where the branch with the `RERAISE` on it can only be reached via a predicate from matching the pending exception against the exception class listed in the `except` clause, and as Numba only supports bare `except` or `except Exception` it renders the `RERAISE` unreachable.

Reply:
It is not reachable at runtime. My intention is to guard against unexpected execution of the RERAISE. Also, we should terminate the BB as byteflow is saying this opcode is a terminator. So, I think raising an Assertion error in `StaticRaise` with a message about the impossible situation makes sense.

Reply:
Done in e94d583, I've also added the standard error reporting text into the `AssertionError` message string. Tested by checking for the python bytecode being present and then checking that the Numba IR has the `StaticRaise` with the right message if so.